### PR TITLE
Sorted by keys

### PR DESCRIPTION
**Pull request by Evgenb000**

### CODE REVIEW

**Review by Zad0rnik — APPROVED** [review]
Good job

File: src/groupByKey.ts [hunk 1]
```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
...
+  return items.reduce((acc, item) => {
+    const groupKey = item[key] as string;
+
+    if (!acc[groupKey]) {
+      acc[groupKey] = [];
+    }
+    acc[groupKey].push(item);
+
+    return acc;
+  }, {} as GroupsMap<T>);
```
Comment (Olexandr6) [inline src/groupByKey.ts:15]: The same here. Declare the initial typed object before using it as the initial value in `reduce`

File: src/groupByKey.ts [hunk 2]
```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
...
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const initialValue: GroupsMap<T> = {};
+
+  return items.reduce((acc, item) => {
+    const groupKey = String(item[key]);
+
+    if (!acc[groupKey]) {
+      acc[groupKey] = [];
+    }
+    acc[groupKey].push(item);
```
Comment (SerhiiKirik) [inline src/groupByKey.ts:14]:
```suggestion
    }

    acc[groupKey].push(item);
```

**Review by SerhiiKirik — APPROVED** [review]
Great 🔥

**Pull request closed without merging**